models/glacier/avalanche_c/glacier_v2_avalanche_c_trades.sql

```diff
@@ -0,0 +1,102 @@
+{{ config(tags=['dunesql'],
+    alias = alias('trades')
+    ,partition_by = ['block_month']
+    ,materialized = 'incremental'
+    ,file_format = 'delta'
+    ,incremental_strategy = 'merge'
+    ,unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index']
+    ,post_hook='{{ expose_spells(\'["avalanche_c"]\',
+                                      "project",
+                                      "glacier",
+                                    \'["discochuck"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2023-03-30' %} -- min(evt_block_time) from glacier_avalanche_c.Pair_evt_Swap
+
+with dexs as (
+    -- glacier_exchange
+    SELECT t.evt_block_time                                             as block_time,
+           t.to                                                         as taker,
+           CAST(NULL as VARBINARY)                                      as maker,
+           case when amount0Out = UINT256 '0' then amount1Out else amount0Out end as token_bought_amount_raw,
+           case when amount0In = UINT256 '0' then amount1In else amount0In end    as token_sold_amount_raw,
+           null                                                          as amount_usd,
+           case when amount0Out = UINT256 '0' then f.token1 else f.token0 end     as token_bought_address,
+           case when amount0In = UINT256 '0' then f.token1 else f.token0 end      as token_sold_address,
+           t.contract_address                                           as project_contract_address,
+           t.evt_tx_hash                                                as tx_hash,
+           t.evt_index
+    FROM
+        {{ source('glacier_avalanche_c', 'Pair_evt_Swap') }} t
+    INNER JOIN {{ source('glacier_avalanche_c', 'PairFactory_evt_PairCreated') }} f
+        on f.pair = t.contract_address
+    {% if is_incremental() %}
+    WHERE t.evt_block_time >= date_trunc('day', now() - interval '7' day)
```

Comment:
We can use `incremental_predicate` to rewrite all incremental filters:
```suggestion
    WHERE {{ incremental_predicate('t.evt_block_time') }}
```

Reply:
This is way above my pay grade (zero). Just trying to help a team I like. I appreciate all the help you and your fellow wizards provide.

Reply:
I will do the incremental filters rewriting, no worries.